src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,286 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+
+
+/**
+ * Represents a translator that makes sense of user input commands.
+ */
+public class Parser {
+    private static String detectedFormat;
+    private static LocalTime startTime;
+    private static LocalTime endTime;
+    private static final String[] dateTimeFormats = {
+        "dd/MM/yyyy HHmm",
+        "dd-MM-yyyy HHmm",
+        "dd/MM/yyyy hh:mm a",
+        "dd-MM-yyyy hh:mm a",
+        "dd/MM/yyyy HHmm",
+        "dd-MM-yyyy HHmm",
+        "yyyy/MM/dd HHmm",
+        "yyyy-MM-dd HHmm",
+        "yyyy/MM/dd hh:mm a",
+        "yyyy-MM-dd hh:mm a",
+        "yyyy/MM/dd HHmm",
+        "yyyy-MM-dd HHmm",
+        "dd MMM yyyy hh:mm a"
+    };
```

Comment:
I like the attention to detail, nice job!

Reply:
Thank you!